templates/base/client/.rtag/State.tsx.hbs

```diff
@@ -0,0 +1,160 @@
+import React, { useRef, useLayoutEffect, useContext } from "react";
+import { RtagContext } from "./context";
+import { RtagConnection } from "./client";
+import {
+  {{userState}} as UserState,
+  {{#each types}}
+  {{@key}},
+  {{/each}}
+} from "./types";
+{{#each plugins}}
+import {{this}}Plugin from "../plugins/{{this}}";
+{{/each}}
+
+{{#each plugins}}
+window.customElements.define("{{makePluginName this}}", {{this}}Plugin);
+{{/each}}
+
+type BaseProps<T> = { value: T };
+
+function KVDisplay(props: { label: string; children: JSX.Element }) {
+  return (
+    <div className="kv-display">
+      <span>{props.label}: </span>
+      {props.children},
+    </div>
+  );
+}
+
+function ArrayDisplay<T>(props: { value: T[]; children: (value: T) => JSX.Element }) {
+  return (
+    <span className="array-display">
+      [
+      {props.value.map((val, i) => (
+        <div className="array-item" key={i}>
+          {props.children(val)}
+        </div>
+      ))}
+      ]
+    </span>
+  );
+}
+
+function OptionalDisplay<T>(props: { value: T | undefined; children: (value: T) => JSX.Element }) {
+  return <span className="optional-display">{props.value !== undefined ? props.children(props.value) : ""}</span>;
+}
+
+function EnumDisplay(props: { value: number; enum: object }) {
+  const labels = Object.entries(props.enum)
+    .filter(([_, value]) => typeof value === "number")
+    .map(([label, _]) => label);
+  return <span className="enum-display">{labels[props.value]}</span>;
```

Comment:
just curious--why span? I feel like i always see div these days

Reply:
`span` is inline unlike div